vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsController.kt

```diff
@@ -152,14 +153,12 @@ class RoomSettingsController @Inject constructor(
                 }
         )
 
-        RoomUtils.getRoomType(roomSummary).let {
-            if (data.actionPermissions.canRemoveFromRoomsDirectory &&
-                    (it == TchapRoomType.FORUM ||
-                            (it == TchapRoomType.UNKNOWN &&
-                                    data.roomDirectoryVisibility is Success &&
-                                    data.roomDirectoryVisibility() == RoomDirectoryVisibility.PUBLIC))) {
-                buildRemoveFromRoomsDirectory()
-            }
+        buildRoomAccessRules(data, roomType)
+
+        if (data.actionPermissions.canRemoveFromRoomsDirectory &&
+                (roomType == TchapRoomType.FORUM ||
+                        (roomType == TchapRoomType.UNKNOWN && data.roomDirectoryVisibility() == RoomDirectoryVisibility.PUBLIC))) {
```

Comment:
Can you please check with @yostyle if he agree to remove the check `data.roomDirectoryVisibility is Success`?
I don't remember why he added it

Reply:
Me too. I don't remember it. You can remove this check.